api/v1alpha1/nodepool_types.go

```diff
@@ -0,0 +1,65 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func init() {
+	SchemeBuilder.Register(&NodePool{})
+	SchemeBuilder.Register(&NodePoolList{})
+}
+
+// NodePool defines the desired state of NodePool
+// +kubebuilder:resource:path=nodepools,shortName=np;nps,scope=Namespaced
+// +kubebuilder:storageversion
+// +kubebuilder:subresource:status
+// +kubebuilder:object:root=true
+type NodePool struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   NodePoolSpec   `json:"spec,omitempty"`
+	Status NodePoolStatus `json:"status,omitempty"`
+}
+
+// NodePoolSpec defines the desired state of NodePool
+type NodePoolSpec struct {
+	// ClusterName is the name of the Cluster this object belongs to.
+	ClusterName string              `json:"clusterName"`
```

Comment:
should likely be of type LocalObjectReference

Reply:
make sense, I was being influenced by https://github.com/kubernetes-sigs/cluster-api/blob/master/api/v1alpha4/machine_types.go#L60